presto-docs/src/main/sphinx/statistics/statistics.rst

```diff
@@ -0,0 +1,155 @@
+statistics information must be available for tables used by query.
+
+Table statistics are provided to query planner by connectors.
+Currently the only connector supporting the mechanism is Hive connector.
+
+Table layouts
+-------------
+
+Statistics are exposed to query planner on per table layout basis. Table layout represents a subset of table's data with attached
+data organization properties (like sort order, bucketing etc.).
+
+How many table layouts table has and details of table layout implementation are specific to connector.
+Taking Hive connector as an example:
+
+* not partitioned tables have just one table layout - representing all data in a table
+* partitioned tables have a family of table layouts. Each subset of partitions to be scanned represent one table layout.
+  Presto will try to pick up smallest possible layout based on filtering predicates from query.
+
+Available statistics
+--------------------
+
+Currently following statics are available in Presto:
+ * **rows count** - total number of rows for table layout
+ * Per each column:
+
+   * **data size** - summary column data size which needs to be read
+   * **nulls count** - number of null values
+   * **distinct values count** - estimated number of distinct values
+
+
+Depending which connector is in use only, subset of statistics may be available.
+For example Hive connector does not support data size for now.
+Also some statistics may be available for one table but not the other,
+or even one table layout and not the other for single table.
+
+Displaying table statistics
+---------------------------
+
+Table statistics can be displayed via Presto CLI using ``SHOW COLUMN STATS`` command.
+
+There are two flavors of the command:
+ * ``SHOW STATS FROM <table_name>`` - will show statistics for table layout representing all data in the table
+ * ``SHOW STATS FROM <table_name> WHERE <filtering_condition> ORDER BY <ordering_expression> LIMIT <limit_count>`` -
+   will show statistics for table layout of table ``t`` representing subset of data after applying given filtering/ordering/limit. Expressions used in
+   ``FILTER`` and ``ORDER BY`` cluase can reference table columns
+
+See the examples output for ``SHOW COLUMN STATS`` command:
+
+.. code-block:: none
+
+    presto:default> SHOW COLUMN STATS FROM nation;
+
+     column_name | data_size | distinct_values_count | nulls_count | row_count
+    -------------+-----------+-----------------------+-------------+-----------
+     comment     | NULL      |                  31.0 |        31.0 | NULL
+     nationkey   | NULL      |                  19.0 |        19.0 | NULL
+     name        | NULL      |                  24.0 |        24.0 | NULL
+     regionkey   | NULL      |                   5.0 |         5.0 | NULL
+     NULL        | NULL      | NULL                  | NULL        |      25.0
+     (5 rows)
+
+
+    presto:default> SHOW COLUMN STATS FROM nation WHERE nationkey > 10 ORDER BY name;
```

Comment:
*This is wrong*, WHERE is used to filter output table not input.
So you can do for example 
SHOW COLUMN STATS FROM nation WHERE nulls_count > 10 ORDER BY name;